src/vos/vos_aggregate.c

```diff
@@ -296,85 +277,54 @@ vos_agg_obj(daos_handle_t ih, vos_iter_entry_t *entry,
 	struct vos_container	*cont = vos_hdl2cont(agg_param->ap_coh);
 	int			 rc;
 
-	if (daos_unit_oid_compare(agg_param->ap_oid, entry->ie_oid)) {
-		if (need_aggregate(agg_param, entry)) {
-			D_DEBUG(DB_EPC, "oid:"DF_UOID" vos agg starting\n",
-				DP_UOID(entry->ie_oid));
-			agg_param->ap_oid = entry->ie_oid;
-			reset_agg_pos(VOS_ITER_DKEY, agg_param);
-			reset_agg_pos(VOS_ITER_AKEY, agg_param);
-			rc = oi_iter_pre_aggregate(ih);
-			if (rc < 0)
-				return rc;
-			if (rc != 0) {
-				/** We removed the key so let's reprobe */
-				*acts |= VOS_ITER_CB_DELETE;
-				inc_agg_counter(cont, VOS_ITER_OBJ, AGG_OP_DEL);
-				return 0;
-			}
-			inc_agg_counter(cont, VOS_ITER_OBJ, AGG_OP_SCAN);
-		} else {
-			D_DEBUG(DB_EPC, "Skip untouched oid:"DF_UOID"\n",
-				DP_UOID(agg_param->ap_oid));
-			*acts |= VOS_ITER_CB_SKIP;
-			inc_agg_counter(cont, VOS_ITER_OBJ, AGG_OP_SKIP);
+	if (need_aggregate(agg_param, entry)) {
+		D_DEBUG(DB_EPC, "oid:"DF_UOID" vos agg starting\n",
+			DP_UOID(entry->ie_oid));
+		agg_param->ap_oid = entry->ie_oid;
```

Comment:
So now we don't require iterator caller to track 'last processed item' anymore, then the ap_oid/dkey/akey could be removed.

Reply:
This is true.  I left it because we have a bunch of DEBUG prints that use them...but we could remove them.